hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/domain/subscription/FilteringServiceSpec.groovy

```diff
@@ -0,0 +1,145 @@
+package pl.allegro.tech.hermes.management.domain.subscription
+
+import pl.allegro.tech.hermes.api.ContentType
+import pl.allegro.tech.hermes.api.MessageFilterSpecification
+import pl.allegro.tech.hermes.api.Topic
+import pl.allegro.tech.hermes.common.filtering.MessageFilters
+import pl.allegro.tech.hermes.common.filtering.avro.AvroPathSubscriptionMessageFilterCompiler
+import pl.allegro.tech.hermes.common.filtering.json.JsonPathSubscriptionMessageFilterCompiler
+import pl.allegro.tech.hermes.management.domain.message.filtering.FilteringConversionException
+import pl.allegro.tech.hermes.management.domain.message.filtering.InvalidFilterTypeException
+import pl.allegro.tech.hermes.api.MessageValidationWrapper
+import pl.allegro.tech.hermes.schema.CompiledSchema
+import pl.allegro.tech.hermes.schema.SchemaRepository
+import pl.allegro.tech.hermes.schema.SchemaVersion
+import pl.allegro.tech.hermes.test.helper.avro.AvroUserSchemaLoader
+import spock.lang.Specification
+
+class FilteringServiceSpec extends Specification {
+
+    MessageFilters messageFilters = new MessageFilters([], [new AvroPathSubscriptionMessageFilterCompiler(), new JsonPathSubscriptionMessageFilterCompiler()])
+    SchemaRepository schemaRepository = Mock()
+    Topic topic = Mock()
+
+    def filteringService = new FilteringService(messageFilters, schemaRepository)
+    def jsonSpec1 = new MessageFilterSpecification([type: "jsonpath", path: ".id", matcher: "0001"])
+    def jsonSpec2 = new MessageFilterSpecification([type: "jsonpath", path: ".name", matcher: "abc"])
+
+    def avroSpec1 = new MessageFilterSpecification([type: "avropath", path: ".id", matcher: "0001"])
+    def avroSpec2 = new MessageFilterSpecification([type: "avropath", path: ".name", matcher: "abc"])
+
+    def schema = AvroUserSchemaLoader.load("/simple.avsc")
+
+    def "should filter json message"() {
+        given:
+        def json = '''
+        {
+            "id": "0001",
+            "name": "XYZ"
+        }
+        '''
+        def wrapper = new MessageValidationWrapper(json, [jsonSpec1, jsonSpec2], null)
+        topic.getContentType() >> ContentType.JSON
+
+        when:
```

Comment:
you can shorten the `when: - then:` here to

    expect:
    filteringService.isFiltered(wrapper, topic).filtered